app/views/users/_extensions.html.erb

```diff
@@ -0,0 +1,53 @@
+<div class="crayons-card crayons-card--content-rows">
+  <h2 class="crayons-subtitle-1">
+    Connect
+  </h2>
+  <%= form_for(@user, html: { id: nil, class: "grid gap-6" }) do |f| %>
+    <div class="crayons-field">
+      <%= f.label :inbox_type, "Inbox type", class: "crayons-field__label" %>
+      <p class="crayons-field__description">Open your inbox to messages from people you don't follow or keep your inbox private to mutual follows.</p>
+      <%= f.select :inbox_type, [%w[Open open], %w[Private private]], {}, { class: "crayons-select" } %>
+    </div>
+
+    <div class="crayons-field">
+      <%= f.label :inbox_guidelines, "Open inbox guidelines/instructions", class: "crayons-field__label" %>
+      <p class="crayons-field__description">
+        Optional
+      </p>
+      <%= f.text_area :inbox_guidelines, maxlength: 200, class: "crayons-textfield", rows: 5, placeholder: "For example:
+  - Only contact me about consulting opportunities.
+  - I'm happy to help beginners with CSS.
+  - Allow for a few days to respond.
+  - etc.
+        " %>
+    </div>
+
+    <button type="submit" class="crayons-btn w-max">
+      Save Connect Settings
+    </button>
+  <% end %>
+</div>
+
+<div class="crayons-card crayons-card--content-rows">
+  <header>
+    <h2 class="crayons-subtitle-1 flex gap-2 items-center">
+      Web monetization
+      <span class="crayons-indicator crayons-indicator--accent">beta</span>
+    </h2>
+    <p class="color-base-70">
+      Receive micropayments for time spent reading your posts directy. <a href="https://dev.to/hacksultan/web-monetization-like-i-m-5-1418">Learn more</a>
+      </p>
+    </p>
+  </header>
+  <%= form_for(@user, html: { id: nil, class: "grid gap-4" }) do |f| %>
+    <div class="crayons-field">
+      <%= f.label :payment_pointer, "Payment Pointer", class: "crayons-field__label" %>
+      <%= f.text_field :payment_pointer, class: "crayons-textfield", placeholder: "$pay.somethinglikethis.co/value" %>
+    </div>
+    <%= f.hidden_field :tab, value: @tab, id: nil %>
+    <div><button type="submit" class="crayons-btn">Save Web Monetization Settings</button></div>
+  <% end %>
+</div>
+
+<%= render partial: "integrations" %>
+<%= render partial: "publishing_from_rss" %>
```

Comment:
I'm adding partials into the views that used to be their own views, but there are 2 things that aren't rendering:

1. Within the existing`integrations` view, the `integrations_github_repositories` isn't rendering.
2. When I try to add `<%= render partial: "response_templates" %>` here, the entire view display an error page.

I'm curious if there is something within those 2 views that need to be specified if I render them as partials. Can you help me here? Thank you!

Reply:
Yes, that's because the logic to make them work is inside the controller. There's a 1:1 correspondence between the name of the section, so for example `extensions` in this case and the logic in the controller:

https://github.com/forem/forem/blob/824e9a13cb1c26d83e23a33f368da5579529e968/app/controllers/users_controller.rb#L231-L246

All the tabs that need special data to be loaded in advance are handled here. I think you can just adapt the `case...when` statement that calls the various functions. Let me know if you need help with that!